posthog-react-native/src/surveys/components/QuestionTypes.tsx

```diff
@@ -253,7 +253,13 @@ export function MultipleChoiceQuestion({
         appearance={appearance}
         onSubmit={() => {
           // If open choice is selected, replace the choice name with the actual value entered
-          onSubmit(selectedChoices.map((c) => (c === openChoice ? openEndedInput : c)))
+          let result: string | string[] = selectedChoices.map((c) => (c === openChoice ? openEndedInput : c))
+
+          if (result.length === 1) {
+            result = result[0]
+          }
```

Comment:
if its a single item, `$survey_response` is a string and not an array with one item, otherwise it breaks the UI